src/app/workflow/launch-third-party/launch-third-party.component.ts

```diff
@@ -1,70 +1,150 @@
-import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { Component, Input, OnChanges, OnDestroy, OnInit, SimpleChanges } from '@angular/core';
 import { MatIconRegistry } from '@angular/material';
 import { DomSanitizer } from '@angular/platform-browser';
 
 import { DescriptorTypeCompatService } from '../../shared/descriptor-type-compat.service';
-import { ToolDescriptor, Workflow, WorkflowVersion } from '../../shared/swagger';
+import { ToolFile, Workflow, WorkflowVersion } from '../../shared/swagger';
 import { WorkflowsService } from '../../shared/swagger/api/workflows.service';
 import { SourceFile } from '../../shared/swagger/model/sourceFile';
-import { LaunchThirdPartyService } from './launch-third-party.service';
+import { GA4GHFilesQuery } from '../../shared/ga4gh-files/ga4gh-files.query';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { DescriptorsQuery } from './state/descriptors-query';
+import { DescriptorsService } from './state/descriptors.service';
+import { DescriptorsStore } from './state/descriptors-store.';
+import { Dockstore } from '../../shared/dockstore.model';
+import { HttpUrlEncodingCodec } from '@angular/common/http';
+import FileTypeEnum = ToolFile.FileTypeEnum;
 
-const importHttpRegEx: RegExp = new RegExp(/^\s*import\s+"https?/, 'm');
+/**
+ *  A component that has buttons linking to external platforms that can launch workflows hosted on Dockstore.
+ *
+ *  ## Instructions for adding a new button
+ *
+ *  1. Update dockstore.model.ts with a new property that has the url of the external platform.
```

Comment:
To be tedious, also update the default URL to compose setup

Reply:
Reminds me that I need to submit a PR myself for compose_setup.